src/lazy/encoder/binary/v1_1/value_writer.rs

```diff
@@ -2454,4 +2521,282 @@ mod tests {
+                ],
+            ),
+            // Struct with multiple FlexSym field names
+            (
+                &[field("foo", "bar"), field("baz", "quux")],
+                &[
+                    // Struct with FlexUInt length
+                    0xFC,
+                    // FlexUInt 18
+                    0x25,
+                    // Enable FlexSym field name encoding
+                    0x00,
+                    // Inline 3-byte field name
+                    // ↓     f     o     o
+                    0xFB, 0x66, 0x6F, 0x6F,
+                    // 3-byte symbol
+                    // ↓     b     a     r
+                    0x93, 0x62, 0x61, 0x72,
+                    // Inline 3-byte field name
+                    // ↓     b     a     z
+                    0xFB, 0x62, 0x61, 0x7A,
+                    // 4-byte symbol
+                    // ↓     q     u     u     x
+                    0x94, 0x71, 0x75, 0x75, 0x78
+                ],
+            ),
+            // Struct with multiple FlexUInt field names followed by a FlexSym field name
+            (
+                &[field(4, "foo"), field(5, "bar"), field("quux", "quuz")],
+                &[
+                    // Struct with FlexUInt length
+                    0xFC,
+                    // FlexUInt length 21 
+                    0x2B,
+                    // FlexUInt symbol ID 4
+                    0x09,
+                    // 3-byte symbol
+                    // ↓     f     o     o
+                    0x93, 0x66, 0x6F, 0x6F,
+                    // FlexUInt symbol ID 5
+                    0x0B,
+                    // 3-byte symbol
+                    // ↓     b     a     r
+                    0x93, 0x62, 0x61, 0x72,
+                    // Enable FlexSym field name encoding
+                    0x00,
+                    // Inline 4-byte field name
+                    // ↓     q     u     u     x
+                    0xF9, 0x71, 0x75, 0x75, 0x78,
+                    // 4-byte symbol
+                    // ↓     q     u     u     z
+                    0x94, 0x71, 0x75, 0x75, 0x7A
+                ],
+            ),
+        ];
+        for (value, expected_encoding) in test_cases {
+            encoding_test(
+                |writer: &mut LazyRawBinaryWriter_1_1<&mut Vec<u8>>| {
+                    writer.write(*value)?;
+                    Ok(())
+                },
+                expected_encoding,
+            )?;
+        }
+        Ok(())
+    }
+
+    #[test]
+    fn write_delimited_structs() -> IonResult<()> {
+        #[rustfmt::skip]
+            let test_cases: &[(TestStruct, &[u8])] = &[
+            // Empty struct
+            (&[], &[0xF3, 0xF0]),
+            // Struct with a single FlexUInt field name
+            (
+                &[field(4, "foo")],
+                &[
+                    // Delimited struct
+                    0xF3,
+                    // FlexUInt symbol ID 4
+                    0x09,
```

Comment:
I think Matt and I were still operating under the assumption that delimited structs always had FlexSym field names: https://github.com/amazon-ion/ion-java/pull/665/files#diff-c491f83b80fd84cfed9e62a411f61789c10d9731d4afbb7fefe538738dd3cd27R407